### ISSUE TITLE
Allow to ignore default arguments when autocompleting single-argument methods

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot-proposals#6733]
### Describe the project you are working on

Game using GDScript.

### Describe the problem or limitation you are having in your project

There are some methods that take a single argument and it has a default value. Every time I complete these methods, I'm left with an opened brace in case I'd liked to use the argument. But I don't. This is especially true in case of `get_children()` and `get_child_count()` since the internal children support was added.

https://user-images.githubusercontent.com/2223172/233742030-8c4d6459-01f9-42cd-9c2f-e9b3fd316324.mp4

I don't have brace auto-closing enabled, but if a method takes no arguments, it's completed with `()`. It's not the case with the methods that take optional argument.

It's mildly irritating, because I use `get_children()` a lot. I don't want to see the optional argument, because I don't care about it at all >_>

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I think there could be an option (editor setting) to ignore such arguments. So if you complete `get_children` method, it's inserted with `()`. If I ever wanted to use the optional argument, I can just press Left and type it. But It has yet to happen.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The setting could be named `text_editor/completion/ignore_single_optional_arguments` or something (although ignore is misleading probably).

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't think so. It's editor behavior.

### Is there a reason why this should be core and not an add-on in the asset library?

Can't be addoned.

**Comment by HolonProduction** [comment]
I don't think an editor setting is needed here. Whether one wants to use optional arguments may vary from case to case. So the question is rather which case is more common.
I believe, that in most cases one does not want to use optional arguments and so in my opinion it would be ok to ignore them always in autocompletion.

Also I don't quite understand why this should be limited to single argument methods. If a method has two optional arguments the editor should behave the same way in my opinion.

**Comment by KoBeWi** [comment]
> Also I don't quite understand why this should be limited to single argument methods.

Well, it could be extended to all arguments probably, but methods with only multiple optional arguments are much less common and I don't remember having similar problems with such method.

**Comment by HolonProduction** [comment]
Yes this methods are not very common with builtin types. I could only find `String.split`. But I believe for the sake of consistency it should behave the same. Also users may declare own methods with a lot of optional parameters.

In addition I tried a quick implementation in https://github.com/godotengine/godot/pull/76340 and it seems easier to implement it in this way. But that should not be a crucial argument.
